src/test/groovy/hudson/plugins/gradle/injection/token/ShortLivedTokenClientTest.groovy

```diff
@@ -120,7 +125,34 @@ class ShortLivedTokenClientTest extends Specification {
         def key = DevelocityAccessCredentials.HostnameAccessKey.of('localhost', 'xyz')
 
         when:
-        def token = new ShortLivedTokenClient().get(mockDevelocity.address.toString(), key, null)
+        def token = new ShortLivedTokenClient(false).get(mockDevelocity.address.toString(), key, null)
+
+        then:
+        token.get().key == 'some-token'
+        token.get().hostname == mockDevelocity.address.host
+    }
+
+    def "get token for an untrusted server"() {
+        given:
+        def mockDevelocity = GroovyEmbeddedApp.of {
+            serverConfig(ServerConfig.builder().tap {
+                def cert = new SelfSignedCertificate('localhost')
+                def sslContext = SslContextBuilder.forServer(cert.certificate(), cert.privateKey())
+                        .sslProvider(SslProvider.JDK)
+                        .build() as SslContext
+                ssl(sslContext)
+            })
+            handlers {
+                post("api/auth/token") {
+                    response.status(200)
+                    response.send('some-token')
+                }
+            }
+        }
+        def key = DevelocityAccessCredentials.HostnameAccessKey.of('localhost', 'xyz')
+
+        when:
+        def token = new ShortLivedTokenClient(true).get(mockDevelocity.address.toString(), key, null)
```

Comment:
Can you please test allowUntrusted = false as well to make sure it fails

Reply:
Agree, added